src/reader/2of5_reader.ts

```diff
@@ -0,0 +1,224 @@
+import BarcodeReader, { Barcode, BarcodeInfo, BarcodePosition } from './barcode_reader';
+
+const N = 1;
+const W = 3;
+const START_PATTERN = [W, N, W, N, N, N];
+const STOP_PATTERN = [W, N, N, N, W];
+const CODE_PATTERN = [
+    [N, N, W, W, N],
+    [W, N, N, N, W],
+    [N, W, N, N, W],
+    [W, W, N, N, N],
+    [N, N, W, N, W],
+    [W, N, W, N, N],
+    [N, W, W, N, N],
+    [N, N, N, W, W],
+    [W, N, N, W, N],
+    [N, W, N, W, N],
+];
+const START_PATTERN_LENGTH = START_PATTERN.reduce((sum, val) => sum + val, 0);
+
+class NewTwoOfFiveReader extends BarcodeReader {
```

Comment:
Why the `New` in the class name `NewTwoOfFiveReader `? Same for the `NewCodabarReader`...

Reply:
yeah, i'll fix that on a future pass .. i'm going through each one, implementing them as a class, copying each function into the class .. named them NewWhatever to start, so my editor doesn't fill with red squiggles .. thanks :-)